### PR TITLE
Save email without confirmation when podmail is disabled.

### PR DESCRIPTION
**Pull request by mahonrig**
Fix and test for issue #7195 

Not entirely sure this is optimal - I kept the first user save so that the validations on the unconfirmed email could run, and then checked podmail status before setting email and saving again.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -203,6 +203,13 @@
         expect(Workers::Mail::ConfirmEmail).to receive(:perform_async).with(@user.id).once
         put(:update, :id => @user.id, :user => { :email => "my@newemail.com"})
       end
+
+      it "saves unconfirmed_email when podmail is disabled" do
+        AppConfig.mail.enable = false
+        put(:update, :id => @user.id, :user => { :email => "my@newemail.com"})
```
Comment (diaspora-code-review) [inline spec/controllers/users_controller_spec.rb]: - Use the new Ruby 1.9 hash syntax.
- Space inside { detected.

Comment (svbergerem) [inline spec/controllers/users_controller_spec.rb]: Please fix this code style issue. The hash syntax should be 

```ruby
put(:update, id: @user.id, user: {...})
```

You can check the code style locally as described here: https://wiki.diasporafoundation.org/Styleguide#Automatic_local_review

Comment (mahonrig) [inline spec/controllers/users_controller_spec.rb]: Want me to fix the rest of the tests in that file? There's a number of them using the older syntax (I had just copied one of the existing tests and modified)

Comment (svbergerem) [inline spec/controllers/users_controller_spec.rb]: For bigger pull requests it's a good idea to keep unrelated changes separate. This makes the changes more readable and makes the review easier.

In this case it's just a small PR, so do whatever you like. :)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -213,9 +213,19 @@ def change_language(user_data)
...
-      @user.send_confirm_email
-      if @user.unconfirmed_email
-        flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+      if AppConfig.mail.enable?
+        @user.send_confirm_email
+        if @user.unconfirmed_email
+          flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+        end
+      else
+        @user.email = @user.unconfirmed_email
```
Comment (SuperTux88) [inline app/controllers/users_controller.rb]: We don't need to save the email to `@user.unconfirmed_email` if mail is disabled on that pod. We can save it directly to `@user.email` then.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (mahonrig) [inline app/controllers/users_controller.rb]: OK, it had looked like there was some useful validations that only ran on the unconfirmed_email field, such as the duplicate email check.

I'll get it changed.

Comment (SuperTux88) [inline app/controllers/users_controller.rb]: The `email` has also a unique index on it, because we need to validate duplicates when somebody registers with an already used mail-address. If there are other checks only on `unconfirmed_email`, we should probably add them to `email` too, because `unconfirmed_email` is only to store the new mail-address temporarily until it is confirmed, and we don't need that when we cant confirm it.

**Review by SuperTux88 — APPROVED** [review]

**Comment by SuperTux88** [comment]
Merged as 5739f6760594b3f153c3d2d11cd81f5cd87a36e8

Thank you 👍
